File: src/main/java/org/tushar/studentmanagement/service/StudentService.java
package org.tushar.studentmanagement.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.tushar.studentmanagement.entity.Student;
import org.tushar.studentmanagement.repository.StudentRepository;

import java.util.List;

@Service
public class StudentService {

    @Autowired
    private StudentRepository studentRepository;

    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    public Student saveStudent(Student student) {
        return studentRepository.save(student);
    }
    public Student getStudentById(Long id) {
        return studentRepository.findById(id).orElse(null);
    }

    public Student updateStudent(Long id, Student student) {
        Student existingStudent = studentRepository.findById(id).orElse(null);

        if (existingStudent != null) {
            existingStudent.setName(student.getName());
            existingStudent.setEmail(student.getEmail());
            existingStudent.setCourse(student.getCourse());

            return studentRepository.save(existingStudent);
        }

        return null;
    }

    public void deleteStudent(Long id) {
        studentRepository.deleteById(id);
    }
}

File: student-frontend/src/services/studentService.js
import api from "./api";

export const getStudents = () => {
    return api.get("/students");
};

export const getStudentById = (id) => {
    return api.get(`/students/${id}`);
};

export const addStudent = (student) => {
    return api.post("/students", student);
};

export const updateStudent = (id, student) => {
    return api.put(`/students/${id}`, student);
};

export const deleteStudent = (id) => {
    return api.delete(`/students/${id}`);
};